doc/development/guide.rst

```diff
@@ -1,279 +1,45 @@
-  Before submitting the PR, you can autoformat your code changes using the
-  `Black <https://github.com/psf/black>`_ Python autoformatter, with max-line length set to 100:
-
-  .. code-block:: bash
-
-      black -l 100 pennylane/path/to/modified/file.py
-
-  We check all of our code against `Pylint <https://www.pylint.org/>`_ for errors.
-  To lint modified files, simply ``pip install pylint``, and then from the source code
-  directory, run
-
-  .. code-block:: bash
-
-      pylint pennylane/path/to/modified/file.py
-
-
-When ready, submit your fork as a `pull request <https://help.github.com/articles/about-pull-requests>`_
-to the PennyLane repository, filling out the pull request template. This template is added
-automatically to the comment box when you create a new issue.
-
-* When describing the pull request, please include as much detail as possible
-  regarding the changes made/new features added/performance improvements. If including any
-  bug fixes, mention the issue numbers associated with the bugs.
-
-* Once you have submitted the pull request, three things will automatically occur:
-
-  - The **test suite** will automatically run on `Travis CI <https://travis-ci.org/XanaduAI/pennylane>`_
-    to ensure that all tests continue to pass.
-
-  - Once the test suite is finished, a **code coverage report** will be generated on
-    `Codecov <https://codecov.io/gh/XanaduAI/pennylane>`_. This will calculate the percentage
-    of PennyLane covered by the test suite, to ensure that all new code additions
-    are adequately tested.
-
-  - Finally, the **code quality** is calculated by
-    `Codefactor <https://app.codacy.com/app/XanaduAI/pennylane/dashboard>`_,
-    to ensure all new code additions adhere to our code quality standards.
-
-Based on these reports, we may ask you to make small changes to your branch before
-merging the pull request into the master branch. Alternatively, you can also
-`grant us permission to make changes to your pull request branch
-<https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/>`_.
-
-.. _adding_new_templates:
-
-.. include:: adding_templates.rst
+Development guide
+=================
+
+The Development guide contains information regarding how to contribute to the
+PennyLane codebase.
+
+The guides below are aimed towards developers and cover how to install
+PennyLane and its dependencies in development mode, run and add tests, write
+documentation, as well as general best practices and an architectural overview
+of PennyLane, its plugins and devices.
+
+.. devicegalleryitem::
+    :name: Installation
+    :description: Dependencies and installation of PennyLane source code using development mode.
+    :link: guide/installation.html
+
+.. devicegalleryitem::
+    :name: Software tests
+    :description: Installing dependencies and running PennyLane tests and coverage.
```

Comment:
(Minor suggestion): small suggestions, was sticking to how [coverage is phrased here](https://pypi.org/project/pytest-cov/)
```suggestion
    :description: Installing dependencies, running the PennyLane test suite and measuring coverage.
```